src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -5885,6 +5763,152 @@ CalcHeatBalanceInsideSurf( Optional_int_const ZoneToResimulate ) // if passed in
 }
 
 void
+TestSurfTempCalcHeatBalanceInsideSurf(
+	Real64 TH12,
+	SurfaceData surface,
```

Comment:
You should pass this by const reference: `SurfaceData const & surface`. This and the other input parameter below are more than likely a majority of slow down. You are passing by copy right now.